src/multiVolumes.cpp

```diff
@@ -126,8 +127,34 @@ void MultiVolumes::carveCuttingMeshes(std::vector<Slicer*>& volumes, const std::
         Slicer& cutting_mesh_volume = *volumes[carving_mesh_idx];
         for (unsigned int layer_nr = 0; layer_nr < cutting_mesh_volume.layers.size(); layer_nr++)
         {
-            Polygons& cutting_mesh_layer = cutting_mesh_volume.layers[layer_nr].polygons;
+            Polygons& cutting_mesh_polygons = cutting_mesh_volume.layers[layer_nr].polygons;
+            Polygons& cutting_mesh_polylines = cutting_mesh_volume.layers[layer_nr].openPolylines;
+            Polygons* cutting_mesh_area = &cutting_mesh_polygons;
+            Polygons cutting_mesh_area_recomputed;
+            coord_t surface_line_width = cutting_mesh.settings.get<coord_t>("wall_line_width_0");
+            { // compute cutting_mesh_area
+                if (cutting_mesh.settings.get<ESurfaceMode>("magic_mesh_surface_mode") == ESurfaceMode::BOTH)
+                {
+                    cutting_mesh_area_recomputed = cutting_mesh_area->unionPolygons(cutting_mesh_polylines.offsetPolyLine(surface_line_width / 2));
+                    cutting_mesh_area = &cutting_mesh_area_recomputed;
+                }
+                else if (cutting_mesh.settings.get<ESurfaceMode>("magic_mesh_surface_mode") == ESurfaceMode::SURFACE)
+                {
+                    // break up polygons into polylines
+                    // they have to be polylines, because they might break up further when doing the cutting
+                    for (PolygonRef poly : cutting_mesh_polygons)
+                    {
+                        poly.add(poly[0]);
+                    }
+                    cutting_mesh_polylines.add(cutting_mesh_polygons);
+                    cutting_mesh_polygons.clear();
+                    cutting_mesh_area_recomputed = cutting_mesh_polylines.offsetPolyLine(surface_line_width / 2);
+                    cutting_mesh_area = &cutting_mesh_area_recomputed;
+                }
+            }
```

Comment:
To make it completely clear that the block is _intended_ to set the `cutting_mesh_area`, I would only initialize `cutting_mesh_area` here.
```suggestion
            Polygons cutting_mesh_area_recomputed;
            Polygons* cutting_mesh_area;
            coord_t surface_line_width = cutting_mesh.settings.get<coord_t>("wall_line_width_0");
            { // compute cutting_mesh_area
                if (cutting_mesh.settings.get<ESurfaceMode>("magic_mesh_surface_mode") == ESurfaceMode::BOTH)
                {
                    cutting_mesh_area_recomputed = cutting_mesh_polygons.unionPolygons(cutting_mesh_polylines.offsetPolyLine(surface_line_width / 2));
                    cutting_mesh_area = &cutting_mesh_area_recomputed;
                }
                else if (cutting_mesh.settings.get<ESurfaceMode>("magic_mesh_surface_mode") == ESurfaceMode::SURFACE)
                {
                    // break up polygons into polylines
                    // they have to be polylines, because they might break up further when doing the cutting
                    for (PolygonRef poly : cutting_mesh_polygons)
                    {
                        poly.add(poly[0]);
                    }
                    cutting_mesh_polylines.add(cutting_mesh_polygons);
                    cutting_mesh_polygons.clear();
                    cutting_mesh_area_recomputed = cutting_mesh_polylines.offsetPolyLine(surface_line_width / 2);
                    cutting_mesh_area = &cutting_mesh_area_recomputed;
                }
                else
                {
                    cutting_mesh_area = &cutting_mesh_polygons;
                }
            }
```